### DISCUSSION TITLE
JVM.dll ISSUE

### DISCUSSION THREAD

**Discussion by sahiljadhav123** [marcelotduarte/cx_Freeze#2473] [Q&A]
Hello Sir, In Other system My build File Showing the Below Error For reading Tabula library what can i Do right now ??? please help me.
ERROR - No JVM shared library file (jvm.dll) found. Try setting up the JAVA_HOME environment variable properly.
Thanks For Reply...

**Comment by marcelotduarte** [comment]
I'm happy to help, but I always need details.
Imagining that it is tabula-py that you use, which depends on another lib in Java, you will have to provide an installation of java/jre on the client (or perhaps embed it in your installation as I see that some systems do) and configure JAVA_HOME.
Assuming you put jre in your directory structure:
```
if getattr(sys, "frozen", False):
    # The application is frozen
    os.environ["JAVA_HOME"] = os.path.join(os.path.dirname(sys.executable), "jre")
```

**Reply by sahiljadhav123** [reply]
i use flask API with tabula for pdf extraction process and when i am creating build file then its read jvm.dll path - C:\Program Files\Java\jdk1.8.0_321\jre\bin\server\jvm.dll  from my system 
and when we use that build into another system then that system need to set same path for jvm.dll so can we have some option to not required to set same path for another system

**Comment by sahiljadhav123** [comment]
java_path = r"D:\Softwares\Java\jre1.8.0_321\bin\server\jvm.dll"
jpype.startJVM(java_path)

I Am using above code its work with default setup path of jvm.dll in application and its run jvm path 
but how can i mentioned into setup file for use only those path for jvm start 
please help sir....
